### PR TITLE
AMAROC-784 update setup-python in pytest workflow

### PR DESCRIPTION
**Pull request by regDaniel**
**Description:**

Please add a brief summary of what this PR does.

**Error(s) fixed:**

Please tag them using `#`, e.g. `Fixed #42`.

**Checklists**:
- [ ] New code includes dedicated tests.
- [ ] New code has been linted.
- [ ] New code follows the project's style.
- [ ] New code is compatible with the 3-Clause BSD license.
- [x] CHANGELOG has been updated.
- [ ] AUTHORS has been updated.
- [ ] Copyright years in module docstrings have been updated.

### CODE REVIEW

**Comment by regDaniel** [comment]
Hi @srethore and @fpavogt this includes @srethore 's fixes on the actions workflow for #121 and adds some more tweaks to make the actions workflow run and not issue any warnings anymore. As there was some trial and error involved and the scope of the changes is quite limited I would suggest to do a squash merge.

File: .github/workflows/CI_pytest.yml [hunk 1]
```diff
@@ -45,22 +46,24 @@ jobs:
         echo $CONDA
 
     - name: Check the setup (II)
-      if: matrix.os != 'windows-latest'
+      if: matrix.os != 'windows-latest' || (matrix.os != 'macos-latest' && matrix.python-version != '3.9')
```
Comment (fpavogt) [inline .github/workflows/CI_pytest.yml:49]: Is the idea to undo these changes down the line ? If yes, then I'm fine with the changes.  If not (i.e. these will stay long-term), then I'd be more in favor of not running a specific configuration at all to start with (rather than skip specific ones at each step).

**Review by fpavogt — APPROVED** [review]
In principle ok with the changes, if they are not meant to stay in the code forever (having all these exceptions at each step is not ideal).

File: .github/workflows/CI_pytest.yml (hunk 1 above)
Comment (regDaniel) [reply .github/workflows/CI_pytest.yml:49]: hmm I think I would keep them as long as we support Python 3.9 as I don't see any sign of the GH actions maintainers to bring back support for Python 3.9 on arm (see https://github.com/actions/setup-python/issues/696). But once Python 3.9 is end-of-life, we can safely remove the lines and the corresponding entry in the test matrix.

Comment (regDaniel) [reply .github/workflows/CI_pytest.yml:49]: Could be temporary though as one of the last contributions of the above issues mentions (https://github.blog/changelog/2024-04-01-macos-14-sonoma-is-generally-available-and-the-latest-macos-runner-image/)

Comment (fpavogt) [reply .github/workflows/CI_pytest.yml:49]: Then I would suggest to do something like [this](https://stackoverflow.com/questions/68994484/how-to-skip-a-configuration-of-a-matrix-with-github-actions):

```
  runs-on: ${{ matrix.os }}
    strategy:
      matrix:
        os: [ubuntu-latest, macos-latest, windows-latest]
        python-version: ['3.9', '3.10', '3.11']
        exclude:
        # excludes python 3.9 on macOS-latest
        - os: macos-latest
           python-version: '3.9'
```

Unless you tried already ?

Comment (regDaniel) [reply .github/workflows/CI_pytest.yml:49]: ah yes, much better, will try!

**Review by fpavogt — APPROVED** [review]

**Review by srethore — APPROVED** [review]
